core/field_angle.ts

```diff
@@ -546,4 +523,41 @@ Css.register(`
 
 fieldRegistry.register('field_angle', FieldAngle);
 
-(FieldAngle.prototype as AnyDuringMigration).DEFAULT_VALUE = 0;
\ No newline at end of file
+(FieldAngle.prototype as AnyDuringMigration).DEFAULT_VALUE = 0;
+
+/**
+ * The two main modes of the angle field.
+ * Compass specifies:
+ *   - clockwise: true
+ *   - offset: 90
+ *   - wrap: 0
+ *   - round: 15
+ *
+ * Protractor specifies:
+ *   - clockwise: false
+ *   - offset: 0
+ *   - wrap: 0
+ *   - round: 15
+ */
+export enum Mode {
+  COMPASS = 'compass',
+  PROTRACTOR = 'protractor',
+}
+
+/**
+ * Extra configuration options for the angle field.
+ */
+export interface Config extends TextConfig {
```

Comment:
FieldAngleConfig? I think in general making these clear/unique would be helpful, and same for the FromJsonConfigs.